src/components/pluginManager.js

```diff
@@ -25,68 +25,85 @@ define(['events'], function (events) {
+                plugin.getRoutes().forEach(function (route) {
+                    definePluginRoute(instance, route, plugin);
+                });
+            }
+
+            if (plugin.type === 'skin') {
+
+                // translations won't be loaded for skins until needed
+                return Promise.resolve(plugin);
+            } else {
+                return new Promise((resolve, reject) => {
+                    loadStrings(plugin)
+                        .then(function () {
+                            resolve(plugin);
+                        })
+                        .catch(reject);
+                });
+            }
+        }
 
-                if (existing) {
-                    resolve(url);
-                    return;
-                }
+        if (typeof pluginSpec === "string") {
+            console.debug('Loading plugin: ' + pluginSpec);
 
-                plugin.installUrl = url;
+            return new Promise(function (resolve, reject) {
+                require([pluginSpec], (pluginFactory) => {
+                    var plugin = new pluginFactory();
 
-                var urlLower = url.toLowerCase();
-                if (urlLower.indexOf('http:') === -1 && urlLower.indexOf('https:') === -1 && urlLower.indexOf('file:') === -1) {
-                    if (url.indexOf(appRouter.baseUrl()) !== 0) {
+                    // See if it's already installed
+                    var existing = instance.pluginsList.filter(function (p) {
+                        return p.id === plugin.id;
+                    })[0];
 
-                        url = appRouter.baseUrl() + '/' + url;
+                    if (existing) {
+                        resolve(pluginSpec);
                     }
-                }
-
-                var separatorIndex = Math.max(url.lastIndexOf('/'), url.lastIndexOf('\\'));
-                plugin.baseUrl = url.substring(0, separatorIndex);
 
-                var paths = {};
-                paths[plugin.id] = plugin.baseUrl;
+                    plugin.installUrl = pluginSpec;
 
-                requirejs.config({
-                    waitSeconds: 0,
-                    paths: paths
-                });
+                    var urlLower = pluginSpec.toLowerCase();
+                    if (urlLower.indexOf('http:') === -1 && urlLower.indexOf('https:') === -1 && urlLower.indexOf('file:') === -1) {
+                        if (pluginSpec.indexOf(appRouter.baseUrl()) !== 0) {
 
-                instance.register(plugin);
+                            pluginSpec = appRouter.baseUrl() + '/' + pluginSpec;
+                        }
+                    }
 
-                if (plugin.getRoutes) {
-                    plugin.getRoutes().forEach(function (route) {
-                        definePluginRoute(instance, route, plugin);
-                    });
-                }
+                    var separatorIndex = Math.max(pluginSpec.lastIndexOf('/'), pluginSpec.lastIndexOf('\\'));
```

Comment:
why `lastIndexOf('\\')`? I mean why care for both slashes?

Reply:
Well I did not add this 🤷🏻‍♀️

Reply:
but you're the one who's changing the code right now ;)

Reply:
Well didn't want to break something so I left everything I didn't understand why it was there

Reply:
Well, okay, let's keep backwards compat here.